cylc/flow/remote.py

```diff
@@ -154,6 +155,66 @@ def construct_platform_ssh_cmd(raw_cmd, platform, **kwargs):
     return ret
 
 
+def get_includes_to_rsync(rsync_includes=None):
+    """Returns a list of directories/files, configured in flow.cylc,
+        to be included in the remote file installation.
+    """
+
+    configured_includes = []
+
+    if rsync_includes:
+        for include in rsync_includes:
+            if include.endswith("/"):  # item is a directory
+                configured_includes.append("/" + include + "***")
+            else:  # item is a file
+                configured_includes.append("/" + include)
+
+    return configured_includes
+
+
+def construct_rsync_over_ssh_cmd(
+        src_path, dst_path, platform, logfile=None, rsync_includes=None):
+    """Constructs the rsync command used for remote file installation
+        Args:
+        src_path(string): source path
+        dst_path(string): path of target
+        platform(dict)): contains info relating to platform
+        logfile(str): the path to the file logging the rsync
+        rsync_includes(list): files and directories to be included in the rsync
+    """
+    dst_host = platform['name']
+    rsync_cmd = "rsync -v --perms --recursive --links --checksum --delete"
+    ssh_cmd = platform['ssh command']
+    rsync_cmd = shlex.split(rsync_cmd)
+    if logfile:
+        rsync_cmd.append(f"--log-file={logfile}")
+    rsync_cmd.append("--rsh=" + ssh_cmd)
+    # Note to future devs - be wary of changing the order of the following
+    # rsync options, rsync is very particular about order of in/ex-cludes.
+
+    rsync_cmd.append('--include=/.service/')
+    rsync_cmd.append('--include=/.service/server.key')
+    excludes = ['log', 'share', 'work']
+    for exclude in excludes:
+        rsync_cmd.append(f"--exclude={exclude}")
+    default_includes = [
+        '/app/***',
```

Comment:
So, rsync patterns are very similar to regexs...
From the man page(https://linux.die.net/man/1/rsync): "if the pattern starts with a / then it is anchored to a particular spot in the hierarchy of files, otherwise it is matched against the end of the pathname." 
So, it's a bit like ^ in regex. We are saying here, include directory that starts with app.

Reply:
Makes sense, thanks for the explanation.

Should we automatically prefix all `default_includes` paths with `/`?

Reply:
So, currently all default_includes have this prefix / added to the paths. In terms of configured_includes, I can add that now.